File: app/Models/Pergunta.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Pergunta extends Model
{
    use HasFactory;

    protected $table = 'perguntas';

    protected $fillable = [
        'formulario_id',
        'numero_da_pergunta',
        'pergunta',
    ];

    public function formulario()
    {
        return $this->belongsTo(Formulario::class, 'formulario_id', 'id');
    }

    public function variaveis()
    {
        return $this->belongsToMany(Variavel::class, 'pergunta_variavel', 'pergunta_id', 'variavel_id');
    }

}

File: migrations/backend-api/app/models/pergunta.py
"""
Modelo Pergunta
"""
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base

class Pergunta(Base):
    __tablename__ = "perguntas"
    
    id = Column(Integer, primary_key=True, index=True)
    formulario_id = Column(Integer, ForeignKey("formularios.id"), nullable=False)
    numero_da_pergunta = Column(Integer, nullable=False)
    pergunta = Column(String(500), nullable=False)
    
    # Timestamps
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    
    # Relationships
    formulario = relationship("Formulario", back_populates="perguntas")
    variaveis = relationship("Variavel", secondary="pergunta_variavel", back_populates="perguntas")
    respostas = relationship("Resposta", back_populates="pergunta", cascade="all, delete-orphan")
    pergunta_variaveis = relationship("PerguntaVariavel", back_populates="pergunta")
